.github/workflows/test.yaml

```diff
@@ -28,6 +28,38 @@ jobs:
       - id: start-time-output
         run: echo "::set-output name=start-time::$(date)"
 
+  # From https://github.com/marketplace/actions/skip-duplicate-actions
+  # This action cleans up previously running instances of a workflow on the same branch.
+  # This accomplishes the task of automatically cancelling CI runs on pushes to the same branch,
+  # which is a common feature in most CI systems but currently not possible with GitHub actions.
+  cleanup-runs:
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    if: "!startsWith(github.ref, 'refs/tags/') && !contains(github.actor, 'renovate') && github.ref != 'refs/heads/dev' && github.ref != 'refs/heads/integration' && github.ref != 'refs/heads/patch-dev' && github.ref != 'refs/heads/latest'"
+    steps:
+      - uses: fkirc/skip-duplicate-actions@v3.4.1
```

Comment:
Can/Should we maybe put this into a separate PR so we can try/merge each in isolation and see what happens?